docs/developer-guide/etendo-classic/how-to-guides/how-to-create-a-standard-process-definition.md

```diff
@@ -0,0 +1,619 @@
+
+###  Uploading files
+
+Process definitions can receive files as parameters. To do that, add a `Process File upload`  Reference as a process parameter. This parameter will show a file upload element in the process form where a single file can be uploaded.
+
+![](../../../assets/developer-guide/etendo-classic/how-to-guides/How_to_create_a_Standard_Process_Definition-20.png)
+
+By default, files uploaded using this component are limited to be 10MB at most. To override this configuration, set the Preference **MaximumProcess File Upload size (MB)** and change this value. File size validation will be performed both client-side and server-side.
+
+Once uploaded a file, file contents and additional data will be available as a parameter entry in the **doExecute(parameters, content)** function in the **Process Event Handler**. The entry corresponding to the file upload will contain a Map with the following information:
+
+```java
+Map<String, Object> params = (Map<String, Object>) parameters.get(paramName);
+ 
+String fileName = params.get(PARAM_FILE_NAME); // The file name
+InputStream content = params.get(PARAM_FILE_CONTENT); // The content of the file as an Stream
+Long size = params.get(PARAM_FILE_SIZE); // The file size in bytes
+```
+
+###  Downloading files
+
+Process definition has the ability to generate and download a file.
+
+In this case of Process Definition, extend from `FileExportActionHandler` and implement the **generateFileToDownload** and **getDownloadFileName** methods.
+
+```java
+/**
+ * Action handler example to export a file from a process definition
+ */
+public class ExportFileExample extends FileExportActionHandler {
+  private static final String FILE_PREFIX = "Test";
+  private static final String FILE_EXTENSION = ".txt";
+ 
+  @Override
+  protected Path generateFileToDownload(Map<String, Object> parameters, JSONObject data)
+      throws IOException, JSONException {
+    String tmpFileName = UUID.randomUUID().toString() + FILE_EXTENSION;
+    File file = new File(ReportingUtils.getTempFolder(), tmpFileName);
+    try (FileWriter outputfile = new FileWriter(file)) {
+      outputfile.write("Hello World!");
+    }
+    return file.toPath();
+  }
+ 
+  @Override
+  protected String getDownloadFileName(Map<String, Object> parameters, JSONObject data) {
+    return FILE_PREFIX + FILE_EXTENSION;
+  }
+}
+```
+
+!!!info
+    If the process definition is launched from a button in a standard window with a header and lines and it is not configured as multi-record, the generated file will be attached to the header by default.
+
+This behavior can be modified overriding the method uploadAttachment.
+
+```java
+@Override
+protected void uploadAttachment(Path originalFile, Map<String, Object> parameters,
+    JSONObject data) throws IOException, JSONException {
+}
+```
+
+!!!info
+    For more information, see an example in [ExportFileExample](https://gitlab.com/users/sign_in){target="\_blank"}.
```

Comment:
@RomanMagnoli Is this link to Gitlab correct?

Reply:
This code must be adapted to compile and uploaded as a zip file.

Reply:
I'll change it because the class that this Java extends doesn't exist anymore... So I'll use a portion of code as example.